### PR TITLE
New Metadata added.

### PR DESCRIPTION
**Pull request by leirbag95**
In order to better filter and to organized my VMs, I added the possibility to give a **name** and a **version tag** to each.

### CODE REVIEW

**Comment by hoh** [comment]
Thanks for your change request.

I see the a few issues that I would like to resolve with you:
1. Making these two fields mandatory would break the validation on existing programs, which is not acceptable for the programs currently running. Marking these fields as `Optional` would solve this.
2. I believe that it would make sense to put these fields together in a sub-field, that could also include tags or other descriptive informations in the future.
3. Can you please submit one clear commit message, in a format similar to the previous commits in the project ?

**Comment by leirbag95** [comment]
Addition of the new `Metadata` field. 
It allows you to add more information to your VMs. 
For the moment we can:
- Name a VM with the `name` field, which makes it easier to identify its programs. The `name` field is a **string**
- Add a `version` to its program. The `version` field is a **string**.

In the future we could add other information like tags, dates or other types of data

File: aleph_message/models/program.py [hunk 1]
```diff
@@ -116,6 +119,7 @@ class ProgramContent(HashableModel, BaseContent):
     type: MachineType = Field(description="Type of execution")
     allow_amend: bool = Field(description="Allow amends to update this function")
     code: CodeContent = Field(description="Code to execute")
+    metadata: Metadata = Field(description="Metadata of the VM")
```
Comment (hoh) [inline aleph_message/models/program.py:122]: I believe this field should be `Optional` to keep backward compatibility.

File: aleph_message/tests/messages/machine_named.json [hunk 2]
```diff
@@ -0,0 +1,116 @@
+{
```
Comment (hoh) [inline aleph_message/tests/messages/machine_named.json:1]: Is this file used anywhere ?

File: aleph_message/models/program.py (hunk 1 above)
Comment (leirbag95) [reply aleph_message/models/program.py]: I replace Metadata class by `metadata: Optional[Dict[str, Any]] = Field(default="Metadata of the VM")`

File: aleph_message/tests/messages/machine_named.json (hunk 2 above)
Comment (leirbag95) [reply aleph_message/tests/messages/machine_named.json:1]: No I didn't use it anywhere and no test was created, so I deleted it

**Comment by hoh** [comment]
Can you please squash all the commits in one ?

File: aleph_message/models/program.py [hunk 3]
```diff
@@ -67,6 +67,9 @@ class FunctionRuntime(HashableModel):
     use_latest: bool = True
     comment: str
 
+class Metadata(HashableModel):
+    name: str
+    version: str
```
Comment (hoh) [inline aleph_message/models/program.py:72]: Why is this still in the PR since it is not used anymore ?

**Pull request merged**